hw/xbox/nv2a/nv2a_debug.c

```diff
@@ -51,34 +65,40 @@ void gl_debug_message(bool cc, const char *fmt, ...)
 
 void gl_debug_group_begin(const char *fmt, ...)
 {
-    size_t n;
-    char buffer[1024];
-    va_list ap;
-    va_start(ap, fmt);
-    n = vsnprintf(buffer, sizeof(buffer), fmt, ap);
-    assert(n <= sizeof(buffer));
-    va_end(ap);
+    /* Debug group begin */
+    if (has_GL_KHR_debug) {
+        size_t n;
+        char buffer[1024];
+        va_list ap;
+        va_start(ap, fmt);
+        n = vsnprintf(buffer, sizeof(buffer), fmt, ap);
+        assert(n <= sizeof(buffer));
+        va_end(ap);
 
-    /* Check for errors before entering group */
-    assert(glGetError() == GL_NO_ERROR);
-
-    if (glPushDebugGroup) {
         glPushDebugGroup(GL_DEBUG_SOURCE_APPLICATION, 0, n, buffer);
     }
+
+    /* Check for errors before starting real commands in group */
+    assert(glGetError() == GL_NO_ERROR);
 }
 
 void gl_debug_group_end(void)
 {
     /* Check for errors when leaving group */
     assert(glGetError() == GL_NO_ERROR);
 
-    if (glPopDebugGroup) {
+    /* Debug group end */
+    if (has_GL_KHR_debug) {
         glPopDebugGroup();
     }
```

Comment:
nit: Used early-outs everywhere else in nv2a_debug.c; this and `gl_debug_initialize` being exceptions. I could negate these 2 condition if requested.